main.cpp

```diff
@@ -50,9 +50,47 @@ struct Wrapper
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
-};
-
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")\n";
+    }
+
+    Type val;
+};
+
+template<>
+struct Wrapper<Point>
+{
+    Wrapper(Point&& t) : val(std::move(t)) 
+    { 
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val.toString() << ")\n";
+    }
+
+    Point val;
+};
```

Comment:
There is a lot of repetition between the template class `Wrapper`, and the template specialization for `Wrapper<Point>`
Can you figure out how to eliminate the duplicate constructor?
hint: you only have to _explicitly specialize_ that which is unique.
You can specialize member functions without specializing the entire class :wink: